### DISCUSSION TITLE
How deepmd generate virial tensor from force and coord caluculated by DFT ?

### DISCUSSION THREAD

**Discussion by ald3hyde** [deepmodeling/deepmd-kit#3156] [Q&A]
Dear everyone,

I do not understand the theory that selecting the virial prefactor for the loss function in DeepMD-kit generates a virial tensor and calculates the virial loss function without virial.npy.

The interatomic force F_ij is needed to calculate the virial tensor. How can this be calculated from F_i, E, and R_i? Is it possible with the formulas in the paper?

I tried "find |grep " to search for a code to calculate the virial tensor from the force (F_i) and the coordinates (coord), but I could not find it, so I would like to know where to find the corresponding code.

Sorry for my lack of study.

**Comment by wanghan-iapcm** [comment]
deepmd does not generate virial tensor from the DFT force and coord. A DFT codes is supposed to calculated the virial by itself.

**Reply by ald3hyde** [reply]
Thank you.

When specifying a virial prefactor other than 0, the virial loss is calculated, and the virial tensor appears to be trained. How does this work? (without virial.npy)
Calculating the loss function despite the absence of a correct (DFT data) label confused me.

**Reply by wanghan-iapcm** [reply]
If no virial label (virial.npy) is provided, then the virial is not trained.

**Reply by njzjz** [reply]
It's just an output issue and has been fixed in #3047.

**Reply by ald3hyde** [reply]
I see. Thank you for your answer.
